exporters/otlp/src/otlp_http_client.cc

```diff
@@ -654,22 +734,38 @@ opentelemetry::sdk::common::ExportResult OtlpHttpClient::Export(
   }
 
   // Send the request
-  auto session = http_client_->CreateSession(options_.url);
-  auto request = session->CreateRequest();
-
-  for (auto &header : options_.http_headers)
   {
-    request->AddHeader(header.first, header.second);
-  }
-  request->SetUri(http_uri_);
-  request->SetTimeoutMs(std::chrono::duration_cast<std::chrono::milliseconds>(options_.timeout));
-  request->SetMethod(http_client::Method::Post);
-  request->SetBody(body_vec);
-  request->ReplaceHeader("Content-Type", content_type);
+    std::lock_guard<std::recursive_mutex> guard{session_manager_lock_};
```

Comment:
Why we need to do all the processing before checking for Shutdown?

Reply:
It's used to protect `isShutdown()`, `is_shutdown_`, `running_sessions_` and `http_client_`. They are all not thread-safe.

Reply:
I think we should make `isShutdown` thread safe, and short circuit here (e.g. return if isShutdown() is true).